cmd/system-probe/modules/ping.go

```diff
@@ -0,0 +1,138 @@
+//go:build linux
+
+package modules
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/DataDog/datadog-agent/cmd/system-probe/api/module"
+	"github.com/DataDog/datadog-agent/cmd/system-probe/config"
+	sysconfigtypes "github.com/DataDog/datadog-agent/cmd/system-probe/config/types"
+	"github.com/DataDog/datadog-agent/cmd/system-probe/utils"
+	pingcheck "github.com/DataDog/datadog-agent/pkg/networkdevice/pinger"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/gorilla/mux"
+	"go.uber.org/atomic"
+	"google.golang.org/grpc"
+)
+
+const (
+	countParam    = "count"
+	intervalParam = "interval"
+	timeoutParam  = "timeout"
+)
+
+type pinger struct{}
+
+// Pinger is a factory for NDMs Ping module
+var Pinger = module.Factory{
+	Name:             config.PingModule,
+	ConfigNamespaces: []string{},
+	Fn: func(cfg *sysconfigtypes.Config) (module.Module, error) {
+		return &pinger{}, nil
+	},
+	NeedsEBPF: func() bool {
+		return false
+	},
+}
+
+var _ module.Module = &pinger{}
+
+func (p *pinger) GetStats() map[string]interface{} {
+	return nil
+}
+
+func (p *pinger) Register(httpMux *module.Router) error {
+	var runCounter = atomic.NewUint64(0)
+
+	httpMux.HandleFunc("/ping/{host}", utils.WithConcurrencyLimit(utils.DefaultMaxConcurrentRequests, func(w http.ResponseWriter, req *http.Request) {
+		start := time.Now()
+		vars := mux.Vars(req)
+		id := getClientID(req)
+		host := vars["host"]
+
+		count, err := getIntParam(countParam, req)
+		if err != nil {
+			log.Errorf("unable to run ping invalid count %s: %s", host, err)
+			w.Write([]byte("invalid count"))
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		interval, err := getIntParam(intervalParam, req)
+		if err != nil {
+			log.Errorf("unable to run ping invalid interval %s: %s", host, err)
+			w.Write([]byte("invalid interval"))
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		timeout, err := getIntParam(timeoutParam, req)
+		if err != nil {
+			log.Errorf("unable to run ping invalid timeout %s: %s", host, err)
+			w.Write([]byte("invalid timeout"))
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		cfg := pingcheck.Config{
+			UseRawSocket: true,
+			Interval:     time.Duration(interval),
+			Timeout:      time.Duration(timeout),
```

Comment:
Are we passing the duration as nanosecond? wondering if we use millisecond instead since the extra precision is not necessary.

Reply:
It's nanosecond precision yes, but when the user configures it in `yaml` it's millisecond precision and we read it in to a `time.Duration`. I just like using `time.Duration` as it's unambiguous on what the units are, it can be easily converted, and it allows us to pass these directly to the ping library we call.

Reply:
+1 to `time.Duration` usage

Reply:
So the int's read above from the query parameters are milliseconds? So this would be `interval * time.Millisecond` for example?

Reply:
Yes, we read it in from the agent config as milliseconds. So a user passes `4000` for example should be 4 seconds. We then multiply it by `time.Millisecond` to convert to a `time.Duration` in nanoseconds. So the units expected for the query param is nanoseconds which makes it easy for us to read these directly in to `time.Duration`